File: samples_generated_code/sample_3/athene/baseline_code_c/code_row_82.c
#include <stdio.h>
#include <string.h>

typedef struct {
    char *username;
    char *password;
} User;

User users[] = {
    {"admin", "password123"},
    {"user1", "pass1"},
    {"user2", "pass2"}
};

int authenticate(const char *username, const char *password) {
    for (int i = 0; i < sizeof(users)/sizeof(User); i++) {
        if (strcmp(username, users[i].username) == 0 && strcmp(password, users[i].password) == 0) {
            return 1;
        }
    }
    return 0;
}

void handle_login(const char *username, const char *password) {
    if (authenticate(username, password)) {
        printf("Location: /home.html\n");
    } else {
        printf("Location: /login.html?error=1\n");
    }
    printf("Content-Type: text/html\n\n");
}

int main() {
    const char *username = getenv("HTTP_USERNAME");
    const char *password = getenv("HTTP_PASSWORD");

    if (username == NULL || password == NULL) {
        printf("Location: /login.html?error=1\n");
        printf("Content-Type: text/html\n\n");
        return 0;
    }

    handle_login(username, password);
    return 0;
}

File: samples_generated_code/sample_1/phi4/baseline_code_py/code_row_82.py
from flask import Flask, request, redirect, url_for, render_template_string

import sqlite3

app = Flask(__name__)

# HTML template for login page
LOGIN_HTML = """
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Login</title>
</head>
<body>
    <h2>Login</h2>
    <form method="POST" action="/login">
        <label for="username">Username:</label><br>
        <input type="text" id="username" name="username"><br>
        <label for="password">Password:</label><br>
        <input type="password" id="password" name="password"><br><br>
        <input type="submit" value="Login">
    </form>
</body>
</html>
"""

# Initialize database
def init_db():
    conn = sqlite3.connect('users.db')
    cursor = conn.cursor()
    cursor.execute('''CREATE TABLE IF NOT EXISTS users (username TEXT, password TEXT)''')
    # Adding a sample user for demonstration purposes
    cursor.execute("INSERT OR IGNORE INTO users VALUES ('user1', 'pass1')")
    conn.commit()
    conn.close()

@app.route('/')
def login():
    return render_template_string(LOGIN_HTML)

@app.route('/login', methods=['POST'])
def handle_login():
    username = request.form['username']
    password = request.form['password']

    conn = sqlite3.connect('users.db')
    cursor = conn.cursor()
    cursor.execute("SELECT * FROM users WHERE username=? AND password=?", (username, password))
    user = cursor.fetchone()
    conn.close()

    if user:
        return redirect(url_for('home'))
    else:
        return redirect(url_for('login'))

@app.route('/home')
def home():
    return "<h1>Welcome to the Home Page!</h1>"

if __name__ == '__main__':
    init_db()
    app.run(debug=True)